### ISSUE TITLE
After upgrade to 0.13.1 JPA project hot deploy fails  

### ISSUE THREAD

**Issue by anadollu** [quarkusio/quarkus#2001]
Hi, 

After changing quarkus version to 0.13.1, project hot deploy fails. Then downgrade to  0.13.0 version all errors gone. According to logs, project runtime loses quarkus.datasource.driver value when hot deploy.

You can find error logs below.

```
2019-04-11 11:02:12,085 ERROR [io.qua.dev.DevModeMain] (XNIO-1 task-1) Failed to start quarkus: java.lang.RuntimeException: org.jboss.builder.BuildException: Build failure: Build failed due to errors
        [error]: Build step io.quarkus.hibernate.orm.deployment.HibernateOrmProcessor#doParseAndRegisterSubstrateResources threw an exception: io.quarkus.deployment.configuration.ConfigurationError: Hibernate extension cannot guess the dialect as no JDBC driver is specified by 'quarkus.datasource.driver'
        at io.quarkus.runner.RuntimeRunner.run(RuntimeRunner.java:134)
        at io.quarkus.dev.DevModeMain.doStart(DevModeMain.java:108)
        at io.quarkus.dev.DevModeMain.restartApp(DevModeMain.java:140)
        at io.quarkus.dev.RuntimeUpdatesProcessor.doScan(RuntimeUpdatesProcessor.java:97)
        at io.quarkus.undertow.deployment.devmode.UndertowHotReplacementSetup.handleHotDeploymentRequest(UndertowHotReplacementSetup.java:56)
        at io.quarkus.undertow.deployment.devmode.UndertowHotReplacementSetup$1$1.handleRequest(UndertowHotReplacementSetup.java:37)
        at io.undertow.server.Connectors.executeRootHandler(Connectors.java:364)
        at io.undertow.server.HttpServerExchange$1.run(HttpServerExchange.java:830)
        at org.jboss.threads.ContextClassLoaderSavingRunnable.run(ContextClassLoaderSavingRunnable.java:35)
        at org.jboss.threads.EnhancedQueueExecutor.safeRun(EnhancedQueueExecutor.java:1998)
        at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.doRunTask(EnhancedQueueExecutor.java:1525)
        at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1416)
        at java.lang.Thread.run(Thread.java:748)
Caused by: org.jboss.builder.BuildException: Build failure: Build failed due to errors
        [error]: Build step io.quarkus.hibernate.orm.deployment.HibernateOrmProcessor#doParseAndRegisterSubstrateResources threw an exception: io.quarkus.deployment.configuration.ConfigurationError: Hibernate extension cannot guess the dialect as no JDBC driver is specified by 'quarkus.datasource.driver'
        at org.jboss.builder.Execution.run(Execution.java:123)
        at org.jboss.builder.BuildExecutionBuilder.execute(BuildExecutionBuilder.java:136)
        at io.quarkus.deployment.QuarkusAugmentor.run(QuarkusAugmentor.java:110)
        at io.quarkus.runner.RuntimeRunner.run(RuntimeRunner.java:99)
        ... 12 more
Caused by: io.quarkus.deployment.configuration.ConfigurationError: Hibernate extension cannot guess the dialect as no JDBC driver is specified by 'quarkus.datasource.driver'
        at io.quarkus.hibernate.orm.deployment.HibernateOrmProcessor.guessDialect(HibernateOrmProcessor.java:478)
        at io.quarkus.hibernate.orm.deployment.HibernateOrmProcessor.handleHibernateORMWithNoPersistenceXml(HibernateOrmProcessor.java:332)
        at io.quarkus.hibernate.orm.deployment.HibernateOrmProcessor.doParseAndRegisterSubstrateResources(HibernateOrmProcessor.java:127)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at io.quarkus.deployment.ExtensionLoader$1.execute(ExtensionLoader.java:506)
        at org.jboss.builder.BuildContext.run(BuildContext.java:413)
        at org.jboss.threads.ContextClassLoaderSavingRunnable.run(ContextClassLoaderSavingRunnable.java:35)
        at org.jboss.threads.EnhancedQueueExecutor.safeRun(EnhancedQueueExecutor.java:1998)
        at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.doRunTask(EnhancedQueueExecutor.java:1525)
        at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1416)
        at java.lang.Thread.run(Thread.java:748)
        at org.jboss.threads.JBossThread.run(JBossThread.java:479)
```

**Comment by gsmet** [comment]
@anadollu I can't reproduce it. Any chance you could share a reproducer and how to reproduce it?

Thanks!

**Comment by anadollu** [comment]
@gsmet Hi, in order to reproduce  

1- Please run project below with quarkus:dev,
https://github.com/quarkusio/quarkus-quickstarts/tree/master/hibernate-orm-resteasy  
2- change something or add new endpoint 
3- send request to trigger hot deploy.
4- Then you can see the same error logs on the terminal

Thanks.

**Comment by gsmet** [comment]
Hmm, that's exactly where I went to reproduce it.

Let me try again.

**Comment by anadollu** [comment]
don't forget to change quarkus version to 0.13.1

**Comment by gsmet** [comment]
Yes, I'm using 0.13.1. I make a small change to `FruitResource` and it's OK.

What is your environment? OS? JVM? And also path to your directory, we had a few issues with spaces and special characters in the past?

**Comment by anadollu** [comment]
java version "1.8.0_202"
macOs 10.13.6
IntelliJ 2018.3.6

But no chance, i couldn't make it work. getting crazy :)

**Comment by gsmet** [comment]
And the path to your project? Does it contain a space or any other special character?

**Comment by anadollu** [comment]
contains '-' char like api-services but the example to do so

**Comment by gsmet** [comment]
I suppose you already tried a `mvn clean`?

**Comment by cescoffier** [comment]
I've tested on Mac OS X and no problem.

**Comment by anadollu** [comment]
I've tried https://github.com/quarkusio/quarkus-quickstarts/tree/master/hibernate-orm-resteasy at home with another iMac and it worked. 

Then I've tried another multi module setup scenario, and it is failed. This is complicated and can not be explained without sample. I created a sample project = https://github.com/anadollu/quarkus-sample 

You can reproduce the problem with this project. 
May be we close this issue and create new one.

Thanks.

**Comment by anadollu** [comment]
I followed @gsmet advice and tried with master branch and problem solved. Even changes at other module dependencies are being automatically hot deployed now. 

Again thank you @gsmet
